deploy/terraform-aws/main.tf

```diff
@@ -282,7 +282,32 @@ resource "aws_iam_role_policy" "k8s_node" {
         "elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
         "iam:CreateServiceLinkedRole",
         "iam:PassRole",
-        "kms:DescribeKey"
+        "kms:DescribeKey",
+        "ssm:DescribeAssociation",
+        "ssm:GetDeployablePatchSnapshotForInstance",
+        "ssm:GetDocument",
+        "ssm:DescribeDocument",
+        "ssm:GetManifest",
+        "ssm:GetParameter",
+        "ssm:GetParameters",
+        "ssm:ListAssociations",
+        "ssm:ListInstanceAssociations",
+        "ssm:PutInventory",
+        "ssm:PutComplianceItems",
+        "ssm:PutConfigurePackageResult",
+        "ssm:UpdateAssociationStatus",
+        "ssm:UpdateInstanceAssociationStatus",
+        "ssm:UpdateInstanceInformation",
+        "ssmmessages:CreateControlChannel",
+        "ssmmessages:CreateDataChannel",
+        "ssmmessages:OpenControlChannel",
+        "ssmmessages:OpenDataChannel",
+        "ec2messages:AcknowledgeMessage",
+        "ec2messages:DeleteMessage",
+        "ec2messages:FailMessage",
+        "ec2messages:GetEndpoint",
+        "ec2messages:GetMessages",
+        "ec2messages:SendReply"
```

Comment:
This role policy is already pretty long since we've mixed permissions needed for Kip with permissions needed for the control plane + workers. How about unwinding it, and split it into 3 parts:
- Permissions for Kip (documented in docs/kip-iam-permissions.md),
- AWS permissions for the control plane + worker permissions from https://github.com/kubernetes/cloud-provider-aws
- SSM from the managed policy

Then we can attach these to the IAM role that we create.